### PR TITLE
fix!: resolve path resolve issues (on windows) and types for virtual modules

### PR DESCRIPTION
**Pull request by sduduzog**
- Addresses https://github.com/tobiasdiez/nuxt-graphql-server/issues/28 by inlining virtual files
- Possibly fixes https://github.com/nuxt/nuxt/issues/19453

This is dependant on: https://github.com/tobiasdiez/nuxt-graphql-server/pull/67 but I didn't want to put all the work into a single PR as this one might have some commentary but shouldn't block bug fixes

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -86,31 +83,32 @@ export default defineNuxtModule<ModuleOptions>({
       },
     })
 
+    setAlias(nuxt, '#graphql/schema', schemaPath)
```
Comment (tobiasdiez) [inline src/module.ts:86]: This line is a duplication of line 66. Is this on purpose?

File: src/module.ts [hunk 2]
```diff
@@ -86,31 +83,32 @@ export default defineNuxtModule<ModuleOptions>({
...
+    // no need to add to nuxt.d.ts as all types are exported from this alias
```
Comment (tobiasdiez) [inline src/module.ts:87]: I don't really understand this comment. Can you expand on it please?

Maybe also move it down to the `prepare:types` hook to describe why we only add the schema and not also resolver to types.

File: src/module.ts [hunk 3]
```diff
@@ -22,24 +22,19 @@ export interface ModuleOptions {
 const logger = useLogger('graphql/server')
 
 function setAlias(nuxt: Nuxt, alias: string, path: string) {
-  // workaround for https://github.com/nuxt/nuxt/issues/19453
-  if (process.env.NODE_ENV === 'development') {
-    // rollup needs a file URL
-    nuxt.options.alias[alias] = pathToFileURL(
-      resolvePath(nuxt.options.buildDir, path),
-    ).href
+  nuxt.hook('nitro:config', nitroConfig => {
```
Comment (tobiasdiez) [inline src/module.ts:25]: Is there a reason why this is change is also needed in production? Please also keep the reference to https://github.com/nuxt/nuxt/issues/19453.

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

**Comment by tobiasdiez** [comment]
@sduduzog now that you other PR is merged, it would be nice if you could have a look at my comments above (and merge in the latest changes)!

**Comment by sduduzog** [comment]
@tobiasdiez  sorry for keeping this held up for so long. I haven't had time to take a look back at it yet. And with the number of comments on the PR, I really want to make it more granular and specific.

Updates to it will be to move anything that has nothing to do with the PR title to another instead.

That will make our discussions precise I believe

**Comment by tobiasdiez** [comment]
No worries! Take your time...

But I don't think the PR is to involved. Once you rebase, it should only contain a couple of changes - so no need to create a new PR in my opinion (but of course its up to you in the end).

**Comment by tobiasdiez** [comment]
Does this PR helps with https://github.com/tobiasdiez/nuxt-graphql-server/issues/71?

@sduduzog I can also take over from here if you are busy the next days. In this case, please shortly reply to my questions above. Thanks!

File: src/module.ts (hunk 3 above)
Comment (sduduzog) [reply src/module.ts:25]: The previous implementation of 'setAlias' added aliases to both vite and nitro. With this change, types are only exposed to nitro. That's where we need access to resolver types, including the schema

File: src/module.ts (hunk 1 above)
Comment (sduduzog) [reply src/module.ts:86]: nope, fixing

**Comment by sduduzog** [comment]
@tobiasdiez While checking out the questions to refresh myself of the context, I ended up resolving some, maybe all pr queries.

Unit tests passing and hopefully the CI will be happy too

**Comment by awdr74100** [comment]
This PR does fix #71. Excellent! Hope it can be released soon.

**Comment by awdr74100** [comment]
Hi @tobiasdiez, can you please confirm this PR? This module is of great help to the project I am currently working on, thank you very much!

**Review by tobiasdiez — APPROVED** [review]
Thanks again!

I mark this as a breaking change, since it's now no longer possible to import the schema from the vue side, but that's also not very important.

**Comment by sduduzog** [comment]
Naturally, I'll be occasionally checking issues to try help with any issues we could have introduced.

**Pull request merged**